### PR TITLE
[fix](Nereids) fix or to in rule

### PR DESCRIPTION
**Pull request by keanji-x**
## Proposed changes

or expression context can't propagation cross or expression.
for example:
```
select (a = 1 or a = 2 or a = 3) + (a = 4 or a = 5 or a = 6)
= select a in [1, 2, 3] + a in [4,5,6]
!= select a in [1, 2, 3] + a in [1, 2, 3, 4, 5, 6]
```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/OrToIn.java [hunk 1]
```diff
@@ -57,7 +55,7 @@
  * adding any additional rule-specific fields to the default ExpressionRewriteContext. However, the entire expression
  * rewrite framework always passes an ExpressionRewriteContext of type context to all rules.
  */
-public class OrToIn extends DefaultExpressionRewriter<OrToInContext> implements
+public class OrToIn extends DefaultExpressionRewriter<ExpressionRewriteContext> implements
```
Comment (jackwener) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/OrToIn.java:58]: `OrToInContext` is useless? Can we remove it?

Comment (XieJiann) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/OrToIn.java:58]: have removed

**Review by wm1581066 — APPROVED** [review]
LGTM

**Review by jackwener — APPROVED** [review]

**Pull request merged**